### DISCUSSION TITLE
support GitHub Enterprise

### DISCUSSION THREAD

**Discussion by anthony-telljohann** [upptime/upptime#875] [Ideas] [answered]
My team would like to use `upptime` with GitHub Enterprise. At the moment, we are unable to use `upptime` because `@upptime/monitor` does not provide a way for users to set the base url for the GitHub API. More specifically, users do not have a way to set the `baseUrl` used by `@octokit/rest`. There may be other changes that will need to be made to support GitHub Enterprise but this is the first road block we hit. We'd be delighted to contribute to `upptime` if the feature request is desired.

**Comment by AnandChowdhary** [comment]
Makes sense! I think we can add this in the [Upptime configuration](https://upptime.js.org/docs/configuration/) to overwrite the API base URL. We currently do this for the frontend in https://upptime.js.org/docs/configuration/#custom-api-base-url so we'll probably have to rename the property for the backend. happy to accept a PR!

**Reply by anthony-telljohann** [reply, marked as answer]
Someone from my team will begin working on the contribution today. It might take a week or two for us to get approval from our company to submit the PR but rest assured it's coming.

**Reply by AnandChowdhary** [reply]
Thanks a ton!

**Reply by chadhutchins182** [reply]
This this support ever happen?
